src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -1984,36 +1985,30 @@ class Animal { }
 ";
             var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
             compilation.VerifyDiagnostics(
-                // (16,22): error CS8780: A variable may not be declared within a 'not' or 'or' pattern.
+                // (19,22): error CS8780: A variable may not be declared within a 'not' or 'or' pattern.
```

Comment:
>A variable may not be declared within a 'not' or 'or' pattern. [](start = 42, length = 62)

This message no longer seems appropriate

Reply:
It is appropriate, as there is only one specific syntactic circumstance in which we permit a pattern variable under a `not`. I don't think the diagnostic needs to explain that.

---
In reply to: [430768074](https://github.com/dotnet/roslyn/pull/44513#discussion_r430768074) [](ancestors = 430768074)